Helper/Cart.php

```diff
@@ -1702,6 +1700,17 @@ public function buildCartFromQuote($quote, $immutableQuote, $items, $placeOrderP
         //Store immutable quote id in metadata of cart
         $cart['metadata']['immutable_quote_id'] = $immutableQuote->getId();
 
+        $cart['metadata'][SessionHelper::ENCRYPTED_SESSION_ID_KEY] = $this->encryptMetadataValue(
```

Comment:
Can we save this under the feature switch?

Reply:
@vitaliyreznikov we might do it but I am not sure if we should. This feature is an improvement of our previous implementation that, in case of missing metadata in update cart, falls back to the old functionality of using cache. So it covers more cases better, except one when we rely on the old code. When missing metadata issue gets fixed in core we should just remove the old code. Looks to me like overuse of feature switches but if you say so I will add it.

Reply:
This is not because of update_cart, I agree we should include metadata into it.
I have two other reasons under my proposal:
- This is an important change so we need feature flags to have the ability to disable it in any cases.
- This is a lot of additional data that will be persisted into our database.

Do I understand correctly we need this PR only for one merchant right now?

Reply:
@vitaliyreznikov fair points, will do.
The update is made per request / issues from one merchant, correct. However, it applies to all the stores / instances that work behind the load balancer.

Reply:
@vitaliyreznikov done